Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,841 @@
+import com.baidu.mapapi.map.BaiduMap;
+import com.baidu.mapapi.map.MapStatus;
+import com.baidu.mapapi.map.Polyline;
+import com.baidu.mapapi.map.PolylineOptions;
+import com.baidu.mapapi.map.Polygon;
+import com.baidu.mapapi.map.PolygonOptions;
+import com.baidu.mapapi.map.MapStatusUpdate;
+import com.baidu.mapapi.map.MapStatusUpdateFactory;
+import com.baidu.mapapi.map.Stroke;
+import com.baidu.mapapi.map.Marker;
+import com.baidu.mapapi.map.MarkerOptions;
+import com.baidu.mapapi.map.BitmapDescriptorFactory;
+import com.baidu.mapapi.model.LatLng;
+import com.baidu.mapapi.map.UiSettings;
+import com.baidu.mapapi.map.MapPoi;
+import com.baidu.mapapi.map.Projection;
+import com.baidu.mapapi.model.LatLngBounds;
+import com.baidu.mapapi.map.MyLocationConfiguration;
+import com.baidu.mapapi.map.MyLocationConfiguration.LocationMode;
+import com.baidu.mapapi.map.MyLocationData;
+
+import com.o3dr.android.client.Drone;
+import com.o3dr.services.android.lib.coordinate.LatLong;
+import com.o3dr.services.android.lib.drone.attribute.AttributeEvent;
+import com.o3dr.services.android.lib.drone.attribute.AttributeType;
+import com.o3dr.services.android.lib.drone.property.FootPrint;
+import com.o3dr.services.android.lib.drone.property.Gps;
+
+import org.droidplanner.android.DroidPlannerApp;
+import org.droidplanner.android.R;
+import org.droidplanner.android.fragments.SettingsFragment;
+import org.droidplanner.android.maps.DPMap;
+import org.droidplanner.android.maps.MarkerInfo;
+import org.droidplanner.android.maps.providers.DPMapProvider;
+import org.droidplanner.android.utils.DroneHelper;
+import org.droidplanner.android.utils.collection.HashBiMap;
+import org.droidplanner.android.utils.prefs.AutoPanMode;
+import org.droidplanner.android.utils.prefs.DroidPlannerPrefs;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.droidplanner.android.maps.providers.google_map.tiles.mapbox.offline.MapDownloader;
+
+public class BaiduMapFragment extends SupportMapFragment implements DPMap{ 
+
+    private static final String TAG = BaiduMapFragment.class.getSimpleName();
+
+    private static final IntentFilter mEventFilter = new IntentFilter();
+    private LocationClient mBDLocClient;
+    public MyBDLocationListenner mBDLocListener = new MyBDLocationListenner();
+    private final HashBiMap<MarkerInfo, Marker> mBiMarkersMap = new HashBiMap<MarkerInfo, Marker>();
+    private DroidPlannerPrefs mAppPrefs;
+    private final AtomicReference<AutoPanMode> mPanMode = new AtomicReference<AutoPanMode>(AutoPanMode.DISABLED);
+
+    private int maxFlightPathSize;
+    private Polyline mFlightPath;
+    private Polyline mMissionPath;
+    private Polyline mDroneLeashPath;
+    public enum PolyLineTypes {
+        LEASH_PATH, MISSION_PATH, FLIGHT_PATH
+    }
```

Comment:
@droneboost let's use android [`IntDef`](https://developer.android.com/reference/android/support/annotation/IntDef.html) here instead of enum. You can see an explanation of its benefits [here](https://noobcoderblog.wordpress.com/2015/04/12/java-enum-and-android-intdefstringdef-annotation/).